ddtrace/internal/utils/http.py

```diff
@@ -158,7 +158,12 @@ def w3c_get_dd_list_member(context):
         tags.append("{}:{}".format(W3C_TRACESTATE_SAMPLING_PRIORITY_KEY, context.sampling_priority))
     if context.dd_origin:
         # the origin value has specific values that are allowed.
-        tags.append("{}:{}".format(W3C_TRACESTATE_ORIGIN_KEY, re.sub(r",|;|=|[^\x20-\x7E]+", "_", context.dd_origin)))
+        tags.append(
+            "{}:{}".format(
+                W3C_TRACESTATE_ORIGIN_KEY, re.sub(r",|;|~|[^\x20-\x7E]+", "_", context.dd_origin).replace("=", "~")
+            )
```

Comment:
something like this might be more readable
```suggestion
        # replace unsupported characters with `_`
        tag_val = re.sub(r",|;|~|[^\x20-\x7E]+", "_", context.dd_origin)
        # Since `=` is an allowed character in datadog tags but not tracestate encode `=` as `~`
        tag_val = tag_val.replace("=", "~")
        tags.append("{}:{}".format(W3C_TRACESTATE_ORIGIN_KEY, tag_val)
```

Reply:
Yeah I couldn't find a way to specify multiple replacements without a lambda for re.sub without using lambda which I don't think is supported in all versions.  I'll break it out into a separate function and make it multi-step for clarity.